app/models/exercise_file.rb

```diff
@@ -55,4 +55,10 @@ def parse_text_data
     self.content = Paperclip.io_adapters.for(attachment.instance.attachment).read
     self.attachment = nil
   end
+
+  def file_type_by_extension(extension)
+    FileType.find_or_create_by(file_extension: extension) do |filetype|
```

Comment:
Should we use `file_type` here as well? Similar to CodeOcean? And should be similar to `spec/factories/file_type.rb`

Reply:
I used `filetype` to avoid a name collision - the method is in exercise_file which has a `belongs_to :file_type`

Reply:
In the factory  I tried to stay coherent

Reply:
Ah, right. Thanks - I didn't see that in the collapsed GitHub view.
I agree with you; we should keep it coherent.